schematics/types/compound.py

```diff
@@ -83,7 +90,11 @@ def native_type(self):
     def fields(self):
         return self.model_class.fields
 
-    def __init__(self, model_spec, **kwargs):
+    def __init__(self,
+                 model_spec,  # type: typing.Type[T]
```

Comment:
It seems that it can't. I've tried various forms like `# type: (typing.Type[T], ...) -> T`: it works for the `model_spec` argument, but breaks other arguments (`**kwargs`, and named parameters).